web/vtctld2/src/app/dashboard/shard.component.ts

```diff
@@ -72,20 +65,55 @@ export class ShardComponent implements OnInit, OnDestroy {
     });
   }
 
+  getActions() {
+    let result: MenuItem[] = [
+      {label: 'Status', items: [
+        {label: 'Validate', command: (event) => {this.openValidateShardDialog(); }},
+        {label: 'Validate Version', command: (event) => {this.openValidateVerShardDialog(); }},
+        {label: 'Shard Replication Positions', command: (event) => {this.openShardReplicationPosDialog(); }},
+      ]},
+      {label: 'Change', items: [
+        {label: 'Delete', command: (event) => {this.openDeleteShardDialog(); }},
+        {label: 'Tablet Externally Reparented', command: (event) => {this.openTabExtRepDialog(); }},
+      ]},
+    ];
+    if (this.featuresService.activeReparents) {
+      result.push(
+        {label: 'Reparent', items: [
+          {label: 'Initialize Shard Master', command: (event) => {this.openInitShardMasterDialog(); }},
+          {label: 'Planned Reparent', command: (event) => {this.openPlanRepShardDialog(); }},
+          {label: 'Emergency Reparent', command: (event) => {this.openEmergencyRepShardDialog(); }},
+        ]}
+      );
+    }
+    return result;
+  }
+
   getTabletActions() {
-    return [
-      {label: 'Delete', command: (event) => {this.openDeleteTabletDialog(); }},
-      {label: 'Ping', command: (event) => {this.openPingTabletDialog(); }},
-      {label: 'Refresh State', command: (event) => {this.openRefreshTabletDialog(); }},
-      {label: 'Set ReadOnly', command: (event) => {this.openSetReadOnlyDialog(); }},
-      {label: 'Set ReadWrite', command: (event) => {this.openSetReadWriteDialog(); }},
-      {label: 'Start Slave', command: (event) => {this.openStartSlaveDialog(); }},
-      {label: 'Stop Slave', command: (event) => {this.openStopSlaveDialog(); }},
-      {label: 'Run Health Check', command: (event) => {this.openRunHealthCheckDialog(); }},
-      {label: 'Ignore Health Error', command: (event) => {this.openIgnoreHealthErrorDialog(); }},
-      {label: 'Demote Master', command: (event) => {this.openDemoteMasterDialog(); }},
-      {label: 'Reparent Tablet', command: (event) => {this.openReparentTabletDialog(); }},
+    let result: MenuItem[] = [
+      {label: 'Status', items: [
+        {label: 'Ping', command: (event) => {this.openPingTabletDialog(); }},
+        {label: 'Refresh State', command: (event) => {this.openRefreshTabletDialog(); }},
+        {label: 'Run Health Check', command: (event) => {this.openRunHealthCheckDialog(); }},
+      ]},
+      {label: 'Change', items: [
+        {label: 'Set ReadOnly', command: (event) => {this.openSetReadOnlyDialog(); }},
+        {label: 'Set ReadWrite', command: (event) => {this.openSetReadWriteDialog(); }},
+        {label: 'Start Slave', command: (event) => {this.openStartSlaveDialog(); }},
+        {label: 'Stop Slave', command: (event) => {this.openStopSlaveDialog(); }},
```

Comment:
These commands are not enabled / disabled by the vtctld flag, so if I made them optional here, it would be inconsistent.

That being said, I don't have a problem revisiting these commands grouping. We'd have to double check they're not used during resharding though (I don't think they are). Let's do that in a different PR though.